core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -311,7 +311,9 @@ private[spark] object UIUtils extends Logging {
       id: Option[String] = None,
       headerClasses: Seq[String] = Seq.empty,
       stripeRowsWithCss: Boolean = true,
-      sortable: Boolean = true): Seq[Node] = {
+      sortable: Boolean = true,
+      // The tooltip information could be None, which indicates header does not have a tooltip.
+      tooltipHeaders: Seq[Option[String]] = Seq.empty): Seq[Node] = {
```

Comment:
Could you add an assertion that if tooltipHeaders are nonEmpty, it is a sequence of the same length as headers? Otherwise, it's easy to add a new column and forget to add/shift the tooltips, and it can go unnoticed, as it's UI and has no automated test.
(same can be done for headerClasses I guess)

Reply:
@planga82 maybe add a `require` here accordingly?

Reply:
Hi @srowen. 
We have put this validation at the thriftServers pages code because we don't want to break  the homogeneous way of treating the parameters at UIUtils.listingTable. headerClasses has no requiere to be the same lenght as headers param so I thought it could be better to do at the same way with the new param and include the validation outside.